### PR TITLE
fix(backup): modify logic to list backup volume for recurring job and CSI snapshotter

### PR DESCRIPTION
**Pull request by mantissahz**
#### Which issue(s) this PR fixes:

Issue # longhorn/longhorn#5411

### CODE REVIEW

File: csi/controller_server.go [hunk 1]
```diff
@@ -270,29 +270,26 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
...
-		return nil, nil
+	for _, bv := range list.Data {
+		if bv.BackupTargetName == vol.BackupTargetName && bv.VolumeName == volumeName {
+			return &bv, nil
+		}
 	}
-
-	return &list.Data[0], nil
+	log.Debugf("cannot find backup volume with backup target %s and volume %s", vol.BackupTargetName, volumeName)
+	return nil, nil
```
Comment (derekbit) [inline csi/controller_server.go:292]: Why not `return nil, fmt.Errorf(....)"

Comment (mantissahz) [reply csi/controller_server.go:292]: As we discussed before, we return the error as nil when we can not find a backup volume as the backup is not completed. https://github.com/longhorn/longhorn-manager/blob/master/csi/controller_server.go#L1247-L1258

**Comment by mantissahz** [comment]
>Can you run the regression test cases involving the changes and paste the links here?

Regression test cases:
- Recurring=Job: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7933/
- CSI: https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7934/

File: csi/controller_server.go (hunk 1 above)
Comment (derekbit) [reply csi/controller_server.go:292]: Oh, I got it. Can you add a comment here.

**Comment by derekbit** [comment]
@mantissahz 
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7933/ is failed

**Comment by mantissahz** [comment]
>@mantissahz
https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7933/ is failed

@derekbit,
The PR ([fix(backup): recurring job test case failed](https://github.com/longhorn/longhorn-tests/pull/2220/commits/1e8f9bae174b68d1b835a08ae5772b0fbc248662)) tries to fix`test_recurring_job.py::test_recurring_jobs_allow_detached_volume` is merged, I will re-run it to check if it still failed.

https://ci.longhorn.io/job/private/job/longhorn-tests-regression/7940/

File: csi/controller_server.go (hunk 1 above)
Comment (mantissahz) [reply csi/controller_server.go]: Added. Thanks

**Review by derekbit — APPROVED** [review]

**Comment by derekbit** [comment]
@mergify backport v1.8.x

**Pull request merged**